tests/spec/test_taskspec.py

```diff
@@ -169,8 +169,15 @@ def test_initialization(spec, expected, tmp_sample_tasks, tmp_imports):
         'extract_product': False,
         'extract_upstream': True
     })
+    task_defaults = {
```

Comment:
glad to see you're adding some tests! a general recommendation: please keep existing tests as they are since we want each test case to only test one thing. make this argument optional so you don't need to modify these tests.

then, create new tests to check the feature you're working on

Reply:
Will do. I modified them since I was not totally sure that it was an optional feature. Thanks for clarifying

Reply:
thanks!